app/components/post/post.js

```diff
@@ -486,175 +243,82 @@ class Post extends PureComponent {
     render() {
         const {
             commentCount,
+            commentedOnDisplayName,
+            commentedOnPost,
             config,
+            displayName,
+            isLastReply,
             post,
             renderReplies,
-            theme
+            theme,
+            user
         } = this.props;
         const style = getStyleSheet(theme);
-        const PROFILE_PICTURE_SIZE = 32;
+        const isPostSystemMessage = isSystemMessage(post);
+        const pendingOrFailed = isPostPendingOrFailed(post);
+        const selected = this.state && this.state.selected ? style.selected : null;
+        const showReplyBar = !!commentedOnPost; //eslint-disable-line no-implicit-coercion
 
-        let profilePicture;
-        let displayName;
-        let messageStyle;
-        if (isSystemMessage(post)) {
-            profilePicture = (
-                <View style={style.profilePicture}>
-                    <MattermostIcon
-                        color={theme.centerChannelColor}
-                        height={PROFILE_PICTURE_SIZE}
-                        width={PROFILE_PICTURE_SIZE}
+        return (
+            <View style={[style.container, this.props.style, selected]}>
+                <View style={[style.profilePictureContainer, (pendingOrFailed && style.pendingPost)]}>
+                    <PostProfilePicture
```

Comment:
Maybe just pass the `postId` and `onViewUserProfile` into this, and just have it look at the store the other props itself? It should be able to figure out the theme and user itself totally fine without being so tightly coupled